From 10f9d412ccaa7e4e29032eb1c8d8f3dde597604c Mon Sep 17 00:00:00 2001
From: David Lechner <david@lechnology.com>
Date: Fri, 20 Sep 2019 15:23:21 -0500
Subject: [PATCH] Reference system packages instead of NuGet packages

In order to build without an Internet connection, we need to reference
the system-provided packages instead of NuGet packages.
---
 KeeAgent/KeeAgent.csproj | 12 +++++-------
 debian/source/format     |  2 +-
 2 files changed, 6 insertions(+), 8 deletions(-)

--- a/KeeAgent/KeeAgent.csproj
+++ b/KeeAgent/KeeAgent.csproj
@@ -12,7 +12,9 @@
     </NuGetPackageImportStamp>
     <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
     <TargetFrameworkProfile />
+    <KeePassPluginDevToolsPath>/usr/lib/keepass2-plugin-dev/</KeePassPluginDevToolsPath>
   </PropertyGroup>
+  <Import Project="/usr/lib/keepass2-plugin-dev/Plgx.targets" />
   <PropertyGroup>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
@@ -179,7 +181,7 @@
       <Project>{9163B6B6-0603-4C6E-BC67-7111ED8CDDE1}</Project>
       <Name>SshAgentLib</Name>
       <PlgxReference>..\SshAgentLib\Ui\WinForms\bin\Release\SshAgentLib.dll</PlgxReference>
-      <PlgxReference>..\SshAgentLib\Ui\WinForms\bin\Release\BouncyCastle.Crypto.dll</PlgxReference>
+      <PlgxReference>..\SshAgentLib\Ui\WinForms\bin\Release\BouncyCastle.dll</PlgxReference>
       <PlgxReference>..\SshAgentLib\Ui\WinForms\bin\Release\Chaos.NaCl.dll</PlgxReference>
     </ProjectReference>
   </ItemGroup>
@@ -204,14 +206,10 @@
     <None Include="Resources\help-3.png" />
   </ItemGroup>
   <ItemGroup>
-    <PackageReference Include="KeePass">
-      <Version>2.43.0.36971</Version>
+    <Reference Include="KeePass">
+      <HintPath>/usr/lib/keepass2/KeePass.exe</HintPath>
       <ExcludeFromPlgx />
-    </PackageReference>
-    <PackageReference Include="PlgxTool">
-      <Version>1.0.0</Version>
-      <ExcludeFromPlgx />
-    </PackageReference>
+    </Reference>
   </ItemGroup>
   <PropertyGroup>
   </PropertyGroup>
@@ -244,7 +242,7 @@
   <Target Name="AfterBuild" Condition="'$(Configuration)' == 'ReleasePlgx'">
     <CallTarget Targets="BuildPlgx" />
     <Delete Files="$(OutDir)Microsoft.WindowsAPICodePack.dll;$(OutDir)Microsoft.WindowsAPICodePack.Shell.dll" />
-    <Delete Files="$(OutDir)SshAgentLib.WinForms.dll;$(OutDir)SshAgentLib.dll;$(OutDir)BouncyCastle.Crypto.dll" />
+    <Delete Files="$(OutDir)SshAgentLib.WinForms.dll;$(OutDir)SshAgentLib.dll;$(OutDir)BouncyCastle.dll" />
     <Delete Files="$(OutDir)Chaos.NaCl.dll;$(OutDir)$(AssemblyName)$(TargetExt)" />
   </Target>
   <ProjectExtensions>
--- a/SshAgentLib/SshAgentLib/SshAgentLib.csproj
+++ b/SshAgentLib/SshAgentLib/SshAgentLib.csproj
@@ -115,8 +115,8 @@
     <Compile Include="OpensshCertificate.cs" />
   </ItemGroup>
   <ItemGroup>
-    <Reference Include="Chaos.NaCl, Version=0.1.0.0, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\dlech.Chaos.NaCl.0.1.0.0\lib\net40\Chaos.NaCl.dll</HintPath>
+    <Reference Include="Chaos.NaCl">
+      <HintPath>/usr/lib/cli/Chaos.NaCl/Chaos.NaCl.dll</HintPath>
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.Data" />
@@ -126,11 +126,9 @@
     <Reference Include="System.Xml" />
     <Reference Include="System.Net" />
     <Reference Include="System.Management" />
-    <Reference Include="Mono.Posix, Version=4.0.0.0, Culture=neutral, PublicKeyToken=0738eb9f132ed756">
-      <HintPath>..\..\packages\Mono.Posix.4.0.0.0\lib\net40\Mono.Posix.dll</HintPath>
-    </Reference>
-    <Reference Include="BouncyCastle.Crypto">
-      <HintPath>..\..\packages\Portable.BouncyCastle.1.8.1.3\lib\net40\BouncyCastle.Crypto.dll</HintPath>
+    <Reference Include="Mono.Posix" />
+    <Reference Include="BouncyCastle">
+      <HintPath>/usr/lib/cli/bccrypto-csharp/BouncyCastle.dll</HintPath>
     </Reference>
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
